### PR TITLE
 Add possibility for docstring display

### LINKED ISSUE #279: Display snippets of documentation alongside selected autocomplete item
### Feature request

Some providers (namely, Haxe compiler, but pretty sure others exist) can be configured to output documentation snippets along with suggestions. I suggest making a widget near the suggestion list that displays documentation for the selected item. Something like this is implemented in Visual Studio:

Screenshot

This can probably be added as an optional element of suggestion object, that can contain HTML-formatted text. For instance, Haxe outputs markdown, which can be crunched with `marked` and sent back as HTML. The problem is, it can be quite large, so it should be scrollable.

### PR DESCRIPTION
**Pull request by tinloaf**
No Markdown rendering yet, since it's not clear which subset of markdown to allow (and how to enforce this).

2015-04-01-214559_1920x1200_scrot
2015-04-01-214859_1920x1200_scrot

Closes #279

### CODE REVIEW

**Comment by basarat** [comment]
Can we have a screenshot?

**Comment by tinloaf** [comment]
Sure, sorry. I should have linked to #279 , where I already posted them. Here you go:

2015-04-01-214559_1920x1200_scrot
2015-04-01-214859_1920x1200_scrot

**Comment by basarat** [comment]
:rose: sweet. I've needed this for atom-typescript

File: styles/autocomplete.less [hunk 1]
```diff
@@ -6,6 +6,20 @@ autocomplete-suggestion-list.select-list.popover-list {
   width: auto;
   display: inline-block;
   min-width: 200px;
+  max-width: 700px;
+
+  .docstring {
```
Comment (benogle) [inline styles/autocomplete.less]: How about calling this `.suggestion-description`?

**Comment by benogle** [comment]
Can you:

* Rebase and use the template approach for creating markup
* Write specs

**Comment by tinloaf** [comment]
Sure thing, I'm only not sure what you mean by 'template approach for creating markup' - can you give me a hint?

**Comment by benogle** [comment]
> can you give me a hint?

Yeah sorry for the lack of context. I've changed the suggestion list from using `document.createElement` to setting innerHTML with a string template. See https://github.com/atom-community/autocomplete-plus/blob/42b98bbfbaadcae1ed09d5c19270f312474467ae/lib/suggestion-list-element.coffee#L137 and https://github.com/atom-community/autocomplete-plus/blob/42b98bbfbaadcae1ed09d5c19270f312474467ae/lib/suggestion-list-element.coffee#L153

**Comment by tinloaf** [comment]
I am moving this to #356 .. I forgot to open a feature branch, and developing in master becomes too annoying. Sorry about that. Please have a look there, I made the requested modifications.

**Pull request closed without merging**